src/partitioning.c

```diff
@@ -261,13 +261,17 @@ ts_partitioning_func_apply(PartitioningInfo *pinfo, Oid collation, Datum value)
 }
 
 TSDLLEXPORT Datum
-ts_partitioning_func_apply_slot(PartitioningInfo *pinfo, TupleTableSlot *slot, bool *isnull)
+ts_partitioning_func_apply_slot(PartitioningInfo *pinfo, TupleTableSlot *slot, bool *isnull,
+								AttrNumber partition_col_idx)
 {
 	Datum value;
 	bool null;
 	Oid collation;
 
-	value = slot_getattr(slot, pinfo->column_attnum, &null);
+	if (partition_col_idx != pinfo->column_attnum)
+		value = slot_getattr(slot, partition_col_idx, &null);
```

Comment:
This line also works for the `else` case, because they are equal, right? So we can remove the `if` and keep this line only.

Reply:
Right, fixed.